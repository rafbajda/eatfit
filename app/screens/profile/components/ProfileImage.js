import React from 'react';
import { Avatar } from 'react-native-elements';
import { globalWhite } from '../../../shared/constants/colors';
import hps from '../utils/helpers';
import { avatarContainerStyle, avatarSize } from '../styles/profileImageStyles';

const defaultAvatar = require('../../../assets/images/default_avatar.png');

const ProfileImage = props => {
    const { changeAvatar, profile } = { ...props };
    const initials = hps.getInitialsFromUser(profile);
    return (
        <Avatar
            onEditPress={changeAvatar}
            containerStyle={avatarContainerStyle}
            title={initials}
            size={avatarSize}
            source={
                profile && profile.photoUrl
                    ? { uri: profile.photoUrl }
                    : defaultAvatar
            }
            editButton={{
                name: 'edit',
                color: globalWhite
            }}
            rounded
            showEditButton
        />
    );
};

export default ProfileImage;
