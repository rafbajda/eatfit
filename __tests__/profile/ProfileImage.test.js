import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import ProfileImage from "../../app/screens/profile/components/ProfileImage";

describe('ProfileImage snapshot', () => {
    jest.useFakeTimers();
    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it('renders component', async () => {
        const tree = renderer.create(<ProfileImage />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
