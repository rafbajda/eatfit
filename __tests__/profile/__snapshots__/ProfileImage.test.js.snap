// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileImage snapshot renders component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "borderRadius": 75,
      "height": 150,
      "marginBottom": 30,
      "marginLeft": "auto",
      "marginRight": "auto",
      "marginTop": 30,
      "width": 150,
    }
  }
  theme={
    Object {
      "colors": Object {
        "disabled": "hsl(208, 8%, 90%)",
        "divider": "#bcbbc1",
        "error": "#ff190c",
        "grey0": "#393e42",
        "grey1": "#43484d",
        "grey2": "#5e6977",
        "grey3": "#86939e",
        "grey4": "#bdc6cf",
        "grey5": "#e1e8ee",
        "greyOutline": "#bbb",
        "platform": Object {
          "android": Object {
            "error": "#f44336",
            "primary": "#2196f3",
            "secondary": "#9C27B0",
            "success": "#4caf50",
            "warning": "#ffeb3b",
          },
          "ios": Object {
            "error": "#ff3b30",
            "primary": "#007aff",
            "secondary": "#5856d6",
            "success": "#4cd964",
            "warning": "#ffcc00",
          },
        },
        "primary": "#2089dc",
        "searchBg": "#303337",
        "secondary": "#8F0CE8",
        "success": "#52c41a",
        "warning": "#faad14",
      },
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "backgroundColor": "#bdbdbd",
        "borderRadius": 75,
        "flex": 1,
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <Image
      onLoad={[Function]}
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": null,
            "width": null,
          },
        ]
      }
      testID="RNE__Image"
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    />
    <View
      accessibilityElementsHidden={true}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#bdbdbd",
            "flex": 1,
            "height": null,
            "justifyContent": "center",
            "width": null,
          }
        }
        testID="RNE__Image__placeholder"
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "height": null,
          "width": null,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#aaa",
        "borderRadius": 25,
        "bottom": 0,
        "height": 50,
        "justifyContent": "center",
        "position": "absolute",
        "right": 0,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 2,
        "width": 50,
      }
    }
  >
    <View
      style={null}
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
          underlayColor="#000"
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
</View>
`;
